deploy_tools/cli.py

```diff
@@ -292,6 +292,55 @@ def deploy(
     click.echo(contract.address)
 
 
+@main.command(
+    short_help="Gives the initcode for a contract, could be used for inclusion in a chain spec"
+)
+@click.argument("contract-name", type=str)
+@click.argument("args", nargs=-1, type=str)
+@contracts_dir_option
+@optimize_option
+@evm_version_option
+@compiled_contracts_path_option
+def initcode(
+    contract_name: str,
+    args: Sequence[str],
+    contracts_dir,
+    optimize,
+    evm_version,
+    compiled_contracts_path: str,
+):
+    """
+    Does not deploy a contract but returns the initcode that should be included in the chain spec to have the contract
+    as a pre-compiled contract
+
+    Pre-compiles a contract with the name CONTRACT_NAME and the constructor arguments ARGS.
+    """
+
+    # We are not going to deploy the contract or interact with the chain.
+    # we just need to build a deploy transaction data
```

Comment:
This is a bit of an overkill to just encode the data. Did you at least try to do it without all that and just encode the data?
Have a look at the code of `buildTransaction` to get an idea.

Reply:
Maybe I misunderstand you, but as said in the first message of the PR, we can "get the bytecode and append the arguments encoded for the constructor."  Which is what I pushed now.

I do not think it is strictly better (or worse) because I knew the initcode is what is in the data of a deployment transaction, but I did not know it was just the bytecode with appended encoded arguments until yesterday. So how the function work seems less obvious.

Reply:
The problem before was that you required a fake web3 object and constructed a hole transaction(with nonce, gas, sender, ...), only to use then the data.

Reply:
Much better now.